src/sentry/integrations/msteams/webhook.py

```diff
@@ -177,29 +202,35 @@ def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
     @transaction_start("MsTeamsWebhookEndpoint")
     def post(self, request: HttpRequest) -> HttpResponse:
         # verify_signature will raise the exception corresponding to the error
-        verify_signature(request)
+        self.verify_webhook_request(request)
 
         data = request.data
         conversation_type = data.get("conversation", {}).get("conversationType")
+        event_type = data["type"]
 
         # only care about conversationUpdate and message
-        if data["type"] == "message":
+        if event_type == "message":
             # the only message events we care about are those which
             # are from a user submitting an option on a card, which
             # will always contain an "payload.actionType" in the data.
             if data.get("value", {}).get("payload", {}).get("actionType"):
+                # Processing card actions can only occur in the Region silo.
+                if SiloMode.get_current_mode() == SiloMode.CONTROL:
+                    return self.respond(status=400)
```

Comment:
I think this is fine for now. Longer term though it would be good to split out the code paths for control and region silos. Ideally there would be a condition early on in the endpoint logic that handles choosing which path to go down.